File: ui/react/src/action/AutomateTable.ts
import TableAutomaton from '../model/TableAutomaton';
import TableAction from './TableAction';
import Action from './Action';

class AutomateTable extends TableAction {
    public automaton: TableAutomaton;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const automaton: TableAutomaton = TableAutomaton.deserialize(input.automaton);
        return new AutomateTable(sheetName, tableName, automaton);
    }

    constructor(sheetName: string, tableName: string, automaton: TableAutomaton) {
        super(sheetName, tableName);
        this.automaton = automaton;
    }

    public wrapper(): Action {
        const action = new Action();
        action.automateTable = this;
        return action;
    }

}

export default AutomateTable;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/AutomateTable.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;
import com.ctrip.ferriswheel.common.table.AutomateConfiguration;

public class AutomateTable extends TableAction implements Action {
    AutomateConfiguration solution;

    public AutomateTable() {
    }

    public AutomateTable(String sheetName, String tableName, AutomateConfiguration solution) {
        super(sheetName, tableName);
        this.solution = solution;
    }

    public AutomateConfiguration getSolution() {
        return solution;
    }

    public void setSolution(AutomateConfiguration solution) {
        this.solution = solution;
    }
}
